gef.py

```diff
@@ -3716,41 +3717,65 @@ def parsed_arglist(arglist):
 
     def synchronize(self):
         """Submit all active breakpoint addresses to IDA/BN"""
-        breakpoints = gdb.breakpoints() or []
-        old_bps = []
-
-        for x in breakpoints:
-            if x.enabled and not x.temporary:
-                val = gdb.parse_and_eval(x.location)
-                addr = str(val).strip().split()[0]
-                addr = long(addr, 16)
-                old_bps.append(addr)
 
         pc = current_arch.pc
+
+        vmmap = get_process_maps()
+        base_address = min([x.page_start for x in vmmap if x.path == get_filepath()])
+        end_address = max([x.page_end for x in vmmap if x.path == get_filepath()])
+        if pc > end_address or pc < base_address:
+            # do not sync in library
+            return
+
+        gdb_bps = []
+        for l in gdb.execute("info break", to_string=True).splitlines():
```

Comment:
basically, try to get the real address of bp
gdb> i b
Num     Type           Disp Enb Address            What
7       breakpoint     keep y   0x00005555555551c2
8       breakpoint     keep y   0x00007ffff7a2ec00 in iconv_open at iconv_open.c:33
here should be 0x00005555555551c2 and 0x00007ffff7a2ec00
then, compute its offset from module and sync them to IDA